File: java-advent-of-code/src/test/java/advent/twenty_eighteen/Day02InventoryManagementSystemTest.java
package advent.twenty_eighteen;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day02InventoryManagementSystemTest {
    @Test
    public void checkReal() {
        Day02InventoryManagementSystem im = new Day02InventoryManagementSystem("/twenty_eighteen/Day02-InventoryManagementSystem-input.txt");

        assertEquals(6225, im.getPart1Answer().intValue());
        assertEquals("revtaubfniyhsgxdoajwkqilp", im.getPart2Answer());
    }

    @Test
    public void checkExample1() {
        Day02InventoryManagementSystem im = new Day02InventoryManagementSystem("/twenty_eighteen/Day02-InventoryManagementSystem-example#1.txt");

        assertEquals(12, im.getPart1Answer().intValue());
        assertEquals("abcde", im.getPart2Answer());
    }

    @Test
    public void checkExample2() {
        Day02InventoryManagementSystem im = new Day02InventoryManagementSystem("/twenty_eighteen/Day02-InventoryManagementSystem-example#2.txt");

        assertEquals(0, im.getPart1Answer().intValue());
        assertEquals("fgij", im.getPart2Answer());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_eighteen/Day02InventoryManagementSystemTest.scala
package advent.twenty_eighteen

import org.scalatest.{FlatSpec, Matchers}

class Day02InventoryManagementSystemTest  extends FlatSpec with Matchers {
  "2018 Day 02 - Input" should "calculate answers" in {
    val dp = new Day02InventoryManagementSystem("/twenty_eighteen/Day02-InventoryManagementSystem-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day02InventoryManagementSystem("/twenty_eighteen/Day02-InventoryManagementSystem-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
